.github/workflows/ci.yml

```
@@ -191,88 +187,44 @@ jobs:
           ./zcutil/build.sh
           -j"${{ steps.nproc.outputs.count }}"
 
-      - name: Authenticate to Google Cloud
-        uses: google-github-actions/auth@v2
-        with:
-          credentials_json: ${{ secrets.GCP_SA_KEY }}
-
-      - name: Upload zcashd artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcashd{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-cli artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-cli{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcashd-wallet-tool artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcashd-wallet-tool{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-inspect artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-inspect{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-tx artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-tx{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload zcash-btest artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/test/test_bitcoin{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src/test
-
-      - name: Upload zcash-gtest artifact
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ${{ format('./src/zcash-gtest{0}', matrix.file_ext) }}
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src
-
-      - name: Upload src/test/buildenv.py
-        uses: google-github-actions/upload-cloud-storage@v2
-        with:
-          path: ./src/test/buildenv.py
-          destination: gh-zcash/${{ github.run_id }}/${{ matrix.name }}/src/test
+      - name: Upload zcashd and related executables (src)
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ format('zcashd{0}', matrix.file_ext) }}-${{ matrix.name }}
```

`matrix.name` is sufficient to gate the binaries, we don't need to complicate the artifact names with file extensions.
```suggestion
          name: zcashd-${{ matrix.name }}
```